### PR TITLE
Split game state schema into multiple schema files

### PR DESCRIPTION
**Pull request by rachidlamouri**
Atomic commits!

- Fixed prettier issues with compiled files
- Split game state schema into multiple files for better scalability
- Added process to automatically denormalize schemas for tools that require normalized schemas
    - We only have to worry about modifying the normalized schemas 

Edit:

- Now depends on #38

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,13 @@
   "version": "0.0.0",
   "scripts": {
     "clean:game": "rm -r saves/*.json",
-    "compile:gameStateSchema": "json2ts src/utils/types/gameState.schema.json > src/utils/types/gameState.d.ts && ts-node scripts/compileRoomTypesTuple.ts",
+    "compile:schemas": "ts-node scripts/denormalizeSchemas.ts && json2ts --cwd=src/utils/schemas/normalized/ src/utils/schemas/normalized/gameState.json > src/utils/types/gameState.d.ts && ts-node scripts/compileRoomTypesTuple.ts && prettier --loglevel silent --write src/utils/types/gameState.d.ts src/utils/types/roomTypesTuple.ts",
```
Comment (jshemas) [inline package.json:6]: Should the `prettier` stuff be it's own command that `compile:schema` calls at the end? That way `compile:schema` doesn't get too long.

Comment (rachidlamouri) [reply package.json:6]: I was thinking of breaking `compile:schema` into several commands.

- denormalize schemas
- schema to typescript
- compile room types tuple
- prettier

I wasn't sure what to call each command

Comment (rachidlamouri) [reply package.json]: updated

File: package.json [hunk 2]
```diff
@@ -3,13 +3,17 @@
...
-    "dev:analyze": "npm run compile:schemas && npm run lint:ts && npm run typecheck",
-    "dev:test": "npm run compile:schemas && npm run test:src && npm run test:e2e",
+    "dev:analyze": "npm run schemas:update && npm run lint:ts && npm run typecheck",
+    "dev:test": "npm run schemas:update && npm run test:src && npm run test:e2e",
     "klicker-knight": "ts-node src/klickerKnight",
     "lint:md": "markdownlint **/*.md --ignore node_modules",
     "lint:ts": "eslint --color --ext .ts,.js .",
     "lint:yaml": "yamllint --ignore=node_modules/**/*.yaml **/*.yaml",
+    "schemas:compile-base-types": "json2ts --cwd=src/utils/schemas/normalized/ src/utils/schemas/normalized/gameState.json > src/utils/types/gameState.d.ts",
+    "schemas:compile-custom-types": "ts-node scripts/compileRoomTypesTuple.ts",
```
Comment (rachidlamouri) [inline package.json:13]: I renamed the script "compileRoomTypesTuple" to "compileCustomTypes" in the next commit

**Review by jshemas — APPROVED** [review]
rip `gameState.schema.json`

**Pull request merged**